docs/book/src/developer/architecture/controllers/multi-tenancy.md

```diff
@@ -11,3 +11,108 @@ In order to support multi tenancy, the following rule applies:
 - Providers SHOULD deploy and run any kind of webhook (validation, admission, conversion)
   following Cluster API codebase best practices for the same release.
 - Providers MUST create and publish a `{type}-component.yaml` accordingly.
+- Providers MUST use the field of name `identityRef` to reference the multi-tenancy CR in the
+  `InfraCluster` CR.
+- Providers MUST implement `identityRef` must be of the following type:
+
+  ```go
+  type ProviderIdentityReference struct {
+    // Kind of the identity.
+    // +kubebuilder:validation:Enum=<Supported provider types>
+    Kind string
+    // Name of the identity.
+    // +kubebuilder:validation:MinLength=1
+    Name string
+  }
+  ```
+
+- A Namespace field MUST never be added to the ProviderIdentityReference type to avoid crossing
+  namespace boundaries.
+
+- Where identity types use private key material, CRs MUST implement a `secretRef` on their spec of
+  type string and only read secrets from the same namespace as the CR for namespaced scope resources
+  OR the controller namespace for cluster-scoped resources.
+
+- Providers MAY support `Secret` as a top-level
```

Comment:
Why not use a single way?

Reply:
@gab-satchi and @yastij were against the indirection.

Reply:
yeah the vsphere account info is just username and password so if we had a CR then it would just have a secret reference and nothing else.

Reply:
Same for CAPO at the moment

Reply:
Did I get it correct that the limitation, when using a Secret as identity type, would be that references to Secret must be in the same namespace?

So directly referencing a secret in the controller namespace (which would be similar to the cluster-wide variant just with Secrets) wouldn't be allowed?

Reply:
I think so, otherwise it's different to how SecretRef works in other contexts like Pods. With a Cluster scoped resource, it's less surprising that secrets have to come from _a_ namespace.

Reply:
Yup sounds good to me. Just wanted to confirm as this requires changes in CAPO (which are fine for me)

Reply:
Does the ability to support a secret at the top level effectively maintain backwards compatibility with what we have today?

Reply:
backward compat would be supplying a nil to `IdentityRef`